tests/test_validate_header.py

```diff
@@ -0,0 +1,174 @@
+import pytest
+
+from eth1spec.base_types import U256, Bytes, Bytes8, Bytes32
+from eth1spec.eth_types import Address, Bloom, Hash32, Header, Root, Uint
+from eth1spec.spec import verify_header
+from tests.helpers import hex2hash
+
+
+def create_fake_parent_header(
```

Comment:
Generally I think we should avoid writing our own tests at all costs. The `ethereum/tests` repo provides some blockchain tests that we could use in place: https://github.com/ethereum/tests/tree/develop/LegacyTests/Constantinople/BlockchainTests

Reply:
It would be good to replace these tests with a runner for some of those tests.

Reply:
I intended to write these as unit tests. IIUC, `ethereum/tests` are integration tests. 
However,  I will replace `create_fake_parent_header `  and `create_fake_header ` by parsing some fixture from `ethereum/tests`.

Reply:
There are some unit tests, like the `TrieTests`. But you're right, they're mostly integration tests. Since this will be "the spec" though, I think we need avoid have any tests that aren't readily available to run on other clients. Ideally, any unit tests we write we should PR into `ethereum/tests` and then test them using that mechanism. If we really need some unit tests for things right now, I suppose we can leave them. But I think our longer-term goal will be to move them to `ethereum/tests`, so I don't want to spend too much time on them.

Reply:
cc: @SamWilsn if you have thoughts on this

Reply:
Personally I'd expect to get ~90% coverage from running `ethereum/tests`, but we might need some unit tests to get to 100%.

I don't have a problem with running Python tests, as long as the vast majority of our coverage comes from `ethereum/tests`. # #